scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/rest/ScheduleRestController_Common_Test.java

```diff
@@ -0,0 +1,295 @@
+public class ScheduleRestController_Common_Test {
+
+	@Autowired
+	private Scheduler scheduler;
+
+	@Autowired
+	private ScheduleDao scheduleDao;
+
+	@Autowired
+	MessageBundleResourceHelper messageBundleResourceHelper;
+
+	@Autowired
+	private WebApplicationContext wac;
+	private MockMvc mockMvc;
+
+	private String appId = TestDataSetupHelper.getAppId_1();
+
+	@Before
+	public void beforeTest() throws SchedulerException {
+		// Clear previous schedules.
+		scheduler.clear();
+
+		mockMvc = MockMvcBuilders.webAppContextSetup(wac).build();
+
+	}
+
+	@After
+	@Transactional
+	public void afterTest() {
+		for (ScheduleEntity entity : scheduleDao.findAllSchedulesByAppId(appId)) {
+			scheduleDao.delete(entity);
+		}
+	}
+
+	@Test
+	@Transactional
+	public void testCreateScheduleWithNoSchedules_Failure_01() throws Exception {
+		// No schedules - null case
+		ObjectMapper mapper = new ObjectMapper();
+		ApplicationScalingSchedules schedules = TestDataSetupHelper
+				.generateSpecificDateSchedulesForScheduleController(appId, 1);
+
+		schedules.setSpecific_date(null);
+
+		String content = mapper.writeValueAsString(schedules);
+
+		ResultActions resultActions = mockMvc
+				.perform(put(getCreateSchedulePath()).contentType(MediaType.APPLICATION_JSON).content(content));
+
+		String errorMessage = messageBundleResourceHelper.lookupMessage("schedule.data.invalid.noSchedules",
+				"app_id=" + appId);
+
+		assertUserError(resultActions, status().isBadRequest(), errorMessage);
+	}
+
+	@Test
+	@Transactional
+	public void testCreateScheduleWithNoSchedules_Failure_02() throws Exception {
+		// No schedules - Empty case
+		ObjectMapper mapper = new ObjectMapper();
+		ApplicationScalingSchedules schedules = TestDataSetupHelper
+				.generateSpecificDateSchedulesForScheduleController(appId, 1);
+
+		schedules.setSpecific_date(Collections.<ScheduleEntity> emptyList());
+
+		String content = mapper.writeValueAsString(schedules);
+
+		ResultActions resultActions = mockMvc
+				.perform(put(getCreateSchedulePath()).contentType(MediaType.APPLICATION_JSON).content(content));
+
+		String errorMessage = messageBundleResourceHelper.lookupMessage("schedule.data.invalid.noSchedules",
+				"app_id=" + appId);
+
+		assertUserError(resultActions, status().isBadRequest(), errorMessage);
+	}
+
+	@Test
+	@Transactional
+	public void testCreateScheduleWithoutAppId_Failure_03() throws Exception {
+
+		ObjectMapper mapper = new ObjectMapper();
+		ApplicationScalingSchedules schedules = TestDataSetupHelper
+				.generateSpecificDateSchedulesForScheduleController(appId, 1);
+		String content = mapper.writeValueAsString(schedules);
+
+		ResultActions resultActions = mockMvc
+				.perform(put("/v2/schedules").contentType(MediaType.APPLICATION_JSON).content(content));
+		resultActions = mockMvc.perform(get("/v2/schedules").contentType(MediaType.APPLICATION_JSON).content(content));
+
+		resultActions.andExpect(status().isNotFound());
```

Comment:
Is the endpoint even getting control? Or is this the framework returning the 404 since it is not matching on any route?